sdk/how-to-guides/versions.md

```diff
@@ -0,0 +1,216 @@
+**Contents:**
+
+- [Set the charm version](#heading--set-the-version-of-the-charm)
+- [Test setting the charm version](#heading--test-setting-the-charm-version)
+- [Set the workload version](#heading--set-the-version-of-the-workload)
+- [Test setting the workload version](#heading--test-setting-the-workload-version)
+
```

Comment:
(1) What would be your suggestion for the title of the entire doc, given these contents? 

(2) The TOC feels like it's need more hierarchy, as the first two titles and the last two each form a story -- the charm version and the workload version story, respectively. 

(3) How does this doc connect to https://juju.is/docs/sdk/workloads and https://juju.is/docs/sdk/interact-with-pebble ? (We don't have to solve this problem now but it feels like a discussion worth having because the workload bit feels like it belongs in the latter docs, but those docs have issues too, and we don't have much on the charm itself from a developer perspective -- that is, nothing beyond the charmcraft side of things.)

Reply:
> (1) What would be your suggestion for the title of the entire doc, given these contents?

What about:

How to > Develop > Set the charm and application version

I'm not sure whether "application version" or "workload version" is better. I think we use the terms fairly interchageably - do you prefer one over the other? The Juju attribute is `workload_version`, but I think people generally understand "application" more than "workload".

> (2) The TOC feels like it's need more hierarchy, as the first two titles and the last two each form a story -- the charm version and the workload version story, respectively.

Yes, there really are two stories here. Another option, if short how-tos are fine, would be that these are two separate docs:

How to > Develop > Set the charm version
How to > Develop > Set the application version

Otherwise, I guess I could just put "Charm version" and "Application version" headings in?

> (3) How does this doc connect to https://juju.is/docs/sdk/workloads and https://juju.is/docs/sdk/interact-with-pebble ?

The charm version part doesn't (which is perhaps another reason it should be split off.

For the workload/application version, it does somewhat relate, but setting the version is the same whether you're writing a machine or k8s charm (except for the event that you're probably going to do it in), and I wouldn't want to repeat the content in both.